p2p/pre2p/raintree/network.go

```diff
@@ -139,6 +148,15 @@ func (n *rainTreeNetwork) networkSendInternal(data []byte, address cryptoPocket.
 }
 
 func (n *rainTreeNetwork) HandleNetworkData(data []byte) ([]byte, error) {
+	blockHeightInt := n.GetBus().GetConsensusModule().GetBlockHeight()
```

Comment:
Persistence should be the source of the current height.

Reply:
I have tried using persistence now but left you a few open questions in discord instead. Once answered we can those changes in.